### DISCUSSION TITLE
Automatic consistent alias

### DISCUSSION THREAD

**Discussion by wilkerlucio** [clj-kondo/clj-kondo#1249] [Ideas]
Hello, the current `:consistent-alias` allows to manual configure the expected. What if it could also infer the alias from reading the sources, and just ensure they always match?

This way we can have the linter to just make sure we re-use the same alias, for every ns.

Probably would be good to have an exception list too to cover edge cases.

**Comment by borkdude** [comment]
@wilkerlucio There is a related issue for clojure LSP here: https://github.com/clojure-lsp/clojure-lsp/issues/388
I think clojure LSP is in a better position to do this since it indexes all of your project automatically.

**Reply by wilkerlucio** [reply]
I understand there is a bit of limitation due to caching, but kondo also already analyses function signatures (that require a wider lookup than the current file), I guess the tradeoff is similar.

In my case I don't use `clojure-lsp`, so that would have to be a new kind of setup, also would be nice to have this running on CI (not sure if I'm missing some kinda of usage of clojure-lsp).
